src/VirtoCommerce.CatalogModule.Core/Services/IDynamicAssociationsValueFetcher.cs

```diff
@@ -0,0 +1,11 @@
+using System.Threading.Tasks;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.DynamicAssociations;
+
+namespace VirtoCommerce.CatalogModule.Core.Services
+{
+    public interface IDynamicAssociationsValueFetcher
```

Comment:
Maybe it could be 
IDynamicAssociationsRuleSelector
{
 Task<DynamicAssociation> GetEffectiveDynamicAssociationRuleAsync(Context, CatalogProduct)
}